pkg/request/pool.go

```diff
@@ -216,39 +226,214 @@ func (rp *Pool) NextRequests(ctx context.Context) [][]byte {
 	return rawRequests
 }
 
-func (rp *Pool) RemoveRequests(requests ...string) error {
-	if atomic.LoadUint32(&rp.batchingEnabled) == 0 {
-		rp.pending.RemoveRequests(requests...)
+func (rp *Pool) RemoveRequests(requestsIDs ...string) error {
+	if !rp.isBatchingEnabled() {
+		rp.pending.RemoveRequests(requestsIDs...)
 		return nil
 	}
 
-	for _, requestID := range requests {
+	for _, requestID := range requestsIDs {
 		rp.batchStore.Remove(requestID)
 	}
 	return nil
 }
 
-// Close removes all the requests, stops all the timeout timers.
+func (rp *Pool) Prune(predicate func([]byte) error) { // TODO pruning after they are batched already might be an issue
```

Comment:
what is the issue?

Reply:
Let's say we prune and many requests got evicted from the pool
Now we might have ready made batches that are actually empty 
What do we think about that?

Reply:
that's correct but i guess we could dequeue them for the batcher and skip them, and for the pending they are naturally cleaned up by GC no?

Reply:
the problem I am raising is that there are requests available but they are batched in newer batches 
so we will be proposing an almost empty batch while there are other requests available